client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -860,6 +863,32 @@ public final Cancellable existsSourceAsync(GetRequest getRequest, RequestOptions
             RestHighLevelClient::convertExistsResponse, listener, emptySet());
     }
 
+    /**
+     * Retrieves the source field only of a document using GetSource API.
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-get.html#_source">Get Source API
+     * on elastic.co</a>
+     * @param getRequest the request
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the response
+     */
+    public RestResponse getSource(GetRequest getRequest, RequestOptions options) throws IOException {
```

Comment:
I think this api needs to use its own Request and Response class.

Although the get source api support almost all request parameters that the get api support, the get source api doesn't support `stored_fields`, `version` and `version_type`. The latter two may be supported in the future, but I don't see `stored_fields` ever be supported (that retrieves something that isn't part of the _source). Also I see that the get api may support `doc_value_fields` option to and this is unrelated to the _source like `stored_fields` is. Therefor I think that this api should have its own high level client side request class.

I also think that this api should have a dedicated response class, that just includes a `Map<String, Object>` field for the source instead of the generic `RestResponse` class.

Reply:
Server's `RestGetSourceAction` returns `BytesRestResponse` object. Actually HLRC getSouce method returns the same, I just use more generic class for the method signature. Will it be ok to replace `RestResponse` with `BytesRestResponse`, or `Map<String, Object>` is preferable anyway?

Reply:
> Server's RestGetSourceAction returns BytesRestResponse object. 

So do many RestAction classes, it is a common way for rest actions to serialize the response to a byte array, so that the networking layer can send a response.

> Will it be ok to replace RestResponse with BytesRestResponse, or Map<String, Object> is preferable anyway?

I think if the getSource(...) methods return a `Map<String, Object>` then that is ok too, because _source is the only thing that this API returns and the `exists(...)` and `existsSource(...)` methods in `RestHighLevelClient` class also return just a `boolean`.

Reply:
@martijnvg thanks for the clarification! Could I use new `GetSourceRequest` class with `exist()` method too as under the hood they use the same server's API? But it's already realised with v6.6.0 (#34519), do we able break compatibility here?

Reply:
Let's do that in a separate PR. We can then deprecate the old methods in favour for the new methods.

Reply:
Ok, I've committed the requested change. Should I create an issue for change `exists(...)` or it's possible to link to this conversation in new PR?

Reply:
No need to create an issue for this. You can just mention it in a new PR.

Reply:
@martijnvg Create PR for that #51789